operator/v1/types_console.go

```diff
@@ -146,6 +146,22 @@ type ConsoleCustomization struct {
 	// +listMapKey=id
 	// +optional
 	Perspectives []Perspective `json:"perspectives"`
+	// consoleURL is an URL to be used as the base console address.
+	// This field is intended for clusters without ingress capability,
+	// where access to routes is not possible.
+	// Invalid value will prevent a console rollout.
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +kubebuilder:validation:XValidation:rule="self.startsWith('https://')",message="console url scheme must be https"
+	ConsoleURL string `json:"consoleURL,omitempty"`
```

Comment:
We probably want to put these new fields on this v1 type behind a featuregate? defer to @JoelSpeed

Reply:
The only 2 cents I can add to this is that the fields are intended to be used on HyperShift managed clusters only. Where disabling of the ingress capability is possible. Other OpenShift form factors won't be impacted. The field descriptions highlight this.

Reply:
I think this should be gated in the first instance. One thing to note is that HyperShift is now able to have fields that do not exist in regular OCP clusters. There is a way to enable a feature gated field only for the HyperShift profile.

I know this was done to enable temporary deltas beteween OCP and HCP, but, I don't know if the plan is to allow those to become permanent or not.

What is the target release here?

Reply:
> I don't know if the plan is to allow those to become permanent or not.

Currently, it's needed on HCP only. There is no use case for the usage of these fields on other form factors.

> What is the target release here?

Well, it was 4.16 but I'm beginning to doubt this :)

Reply:
>Currently, it's needed on HCP only. There is no use case for the usage of these fields on other form factors.

David is out this week so I can't confirm with him til next week.

We are now at the `acknowledge-critical-fixes` only stage of 4.16 so I think it's probably not going to make 4.16 unless you want to claim this is critical (which I guess it could be? That's your call 👀 )

Reply:
@jhadvig: what do you think about the feature gate? If you think it's needed I'll postpone the change to 4.17 as I won't have time to think this thoroughly.